tools/gluten-it/package/pom.xml

```diff
@@ -17,8 +17,50 @@
       <groupId>io.glutenproject</groupId>
       <artifactId>gluten-it-common</artifactId>
       <version>${gluten.version}</version>
+      <exclusions>
+        <exclusion>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.trino.tpch</groupId>
+          <artifactId>tpch</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.trino.tpcds</groupId>
+          <artifactId>tpcds</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-repl_${scala.binary.version}</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-catalyst_${scala.binary.version}</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-hive_${scala.binary.version}</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <type>test-jar</type>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-catalyst_${scala.binary.version}</artifactId>
+      <type>test-jar</type>
```

Comment:
Any reason to add the dependencies?

Reply:
In case of possible mismatching between our understandings on the module relationship of `gluten-it`, let's clarify it:

1. `gluten-it-common`: This module includes all the jars needed by gluten-it
2. `gluten-it-package`: This module only copies the jars to `target/libs`

Since `gluten-it` doesn't use a fat jar to do its jobs, is this PR going to put a shaded version of guava to `target/libs`?

If the answer is yes, the PR makes sense to me.

By the way, would you like to share the error log from `gluten-it` about this issue from your environment?

Reply:
thank you for the clarification! yes, that's what this pr did.

The gluten-it lib after this pr:
![image](https://github.com/oap-project/gluten/assets/12025282/0135ebc2-9df1-41ba-b7f1-a95d6a8c8c06)

The error msg:
```
Successfully switched to SessionDesc(SessionToken(baseline),Data Gen) session. 
Generating table lineitem...
23/07/17 17:30:16 ERROR Executor: Exception in task 0.0 in stage 0.0 (TID 0)
java.lang.IncompatibleClassChangeError: Class com.google.common.base.Suppliers$NonSerializableMemoizingSupplier does not implement the requested interface java.util.function.Supplier
	at io.trino.tpch.Distributions.getDefaultDistributions(Distributions.java:48)
	at io.trino.tpch.LineItemGenerator.<init>(LineItemGenerator.java:62)
	at io.glutenproject.integration.tpc.h.TpchDataGen.$anonfun$lineItemGenerator$1(TpchDataGen.scala:48)
	at io.glutenproject.integration.tpc.h.TpchDataGen.$anonfun$lineItemGenerator$1$adapted(TpchDataGen.scala:47)
	at io.glutenproject.integration.tpc.h.TpchDataGen.$anonfun$generate$2(TpchDataGen.scala:310)
	at org.apache.spark.sql.execution.MapPartitionsExec.$anonfun$doExecute$3(objects.scala:201)
```

Reply:
@ulysses-you Thanks for providing the information.

However can we still put all jars to `target/lib` rather than generating a single one? Since it's preferred not to touch the jar we are going to test against, which is `gluten-package-xxx.jar` in this case.

Will it work to exclude guava jar in trino jars from gluten-it directly?

Reply:
> However can we still put all jars to target/lib rather than generating a single one?

This pr just shades gluten-it-common.jar/trino-tpcds/trino-tpcd/guava into one jar. Others are same as before.
![image](https://github.com/oap-project/gluten/assets/12025282/dac52c60-dd69-4cb1-be6b-4c7dcc392e53)

> Will it work to exclude guava jar in trino jars from gluten-it directly?

I'm not sure, but I think it's risky whatever it can work or not..

Reply:
Thanks for clarifying, so the lib folder still includes all the jars separately. That makes sense to me.

In addition, if using this approve, could you please try moving dependency 'gluten-package' from `gluten-it-common` to `gluten-it-package` too? So that we have all the jars we are testing against made as provided/runtime dependency of `gluten-it-package`.

Reply:
sure, addressed